support/cas-server-support-consent-webflow/src/main/java/org/apereo/cas/web/consent/config/CasConsentOverviewConfiguration.java

```diff
@@ -0,0 +1,107 @@
+import org.pac4j.springframework.web.SecurityInterceptor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+import javax.annotation.PostConstruct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is {@link CasConsentOverviewConfiguration}.
+ *
+ * @author Arnold Bergner
+ * @since 5.2.0
+ */
+@Configuration("casConsentOverviewConfiguration")
+public class CasConsentOverviewConfiguration extends WebMvcConfigurerAdapter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CasConsentOverviewConfiguration.class);
+    
+    private static final String CAS_CLIENT_NAME = "CasClient";
+    
+    @Autowired
+    private CasConfigurationProperties casProperties;
+    
+    @Autowired
+    @Qualifier("servicesManager")
+    private ServicesManager servicesManager;
+
+    @Autowired
+    @Qualifier("webApplicationServiceFactory")
+    private ServiceFactory<WebApplicationService> webApplicationServiceFactory;
+
+    @Bean
+    public CasConsentOverviewController casConsentOverviewController() {
+        return new CasConsentOverviewController();
+    }
+    
+    @Bean
+    public IsAuthenticatedAuthorizer isAuthenticatedAuthorizer() {
+        return new IsAuthenticatedAuthorizer();
+    }
+    
+    @Bean
+    public SecurityInterceptor casConsentOverviewSecurityInterceptor() {        
+        return new CasConsentOverviewSecurityInterceptor(config(), CAS_CLIENT_NAME,
+                "securityHeaders,csrfToken,".concat(IsAuthenticatedAuthorizer.class.getSimpleName()));
+    }
+    
+    /**
+    * Initialize consent service.
+    */
+    @PostConstruct
+    protected void registerConsentService() {
+        final Service callbackService = this.webApplicationServiceFactory.createService(
+                casProperties.getServer().getPrefix().concat("/consent.*"));
+        if (!this.servicesManager.matchesExistingService(callbackService)) {
+            LOGGER.debug("Initializing consent service [{}]", callbackService);
+
+            final RegexRegisteredService service = new RegexRegisteredService();
+            service.setId(Math.abs(RandomUtils.getInstanceNative().nextLong()));
+            service.setEvaluationOrder(0);
+            service.setName(service.getClass().getSimpleName());
+            service.setDescription("CAS Consent Overview");
+            service.setServiceId(callbackService.getId());
+
+            LOGGER.debug("Saving consent service [{}] into the registry", service);
+            this.servicesManager.save(service);
+            this.servicesManager.load();
+        }
+    }
+    
+    public Config config() {
```

Comment:
Should probably be a `@Bean`, and probably deserves a better name to avoid conflicts.